airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -472,3 +464,196 @@ class Labels(GoogleAdsStream):
     """
 
     primary_key = ["label.id"]
+
+
+class ChangeStatus(IncrementalGoogleAdsStream):
+    """
+    Change status stream: https://developers.google.com/google-ads/api/fields/v14/change_status
+    Stream is only used internally to implement incremental updates for child streams of IncrementalEventsStream
+    """
+
+    cursor_field = "change_status.last_change_date_time"
+    range_days = 1
+    days_of_data_storage = 90
+
+    def __init__(self, **kwargs):
+        # date range is not used for these streams, only state is used to sync recent records, otherwise full refresh
+        for key in ["start_date", "conversion_window_days", "end_date"]:
+            kwargs.pop(key, None)
+        super().__init__(start_date=None, conversion_window_days=1, end_date=None, **kwargs)
+
+    def get_query(self, stream_slice: Mapping[str, Any]) -> str:
+        query = GoogleAds.convert_schema_into_query(
+            schema=self.get_json_schema(),
+            report_name=self.name,
+            from_date=stream_slice.get("start_date"),
+            to_date=stream_slice.get("end_date"),
+            cursor_field=self.cursor_field,
+            # resource type is used for filtering updates only for desirable stream
+            resource_type=stream_slice.get("resource_type"),
+            # limit is mandatory parameter for this stream, also 10k is maximum value
+            limit=10000,
+        )
+        return query
+
+
+class IncrementalEventsStream(GoogleAdsStream, IncrementalMixin, ABC):
```

Comment:
this should follow the same pattern for deletions as shopify, could you sync with @bazarnov about how to do that?

Reply:
@sherifnada No, not all resources from Change Status should be affected by the new class.
We already fetch incremental updates for resources like Campaigns, Ad Groups, Ad Group Ads, and Campaign Budget based on date.
Also, these resources, unlike criterions, can't be deleted, only marked as "Removed". Our approach captures this.
Given these points, using ChangeStatus for all resources isn't necessary. Our method ensures we capture all relevant updates efficiently.

Reply:
Great, thanks for the explanation! Can you add this context to the docstring for this class? For the next developer reading this, it wouldn't be obvious why we're using this mechanism for two streams only so it is helpful to retain this knowledge